docs/docs/Operators.md

```diff
@@ -0,0 +1,208 @@
+(push-this)       ;; ( )
+```
+
+`this` is a register that refers to the current object.
+
+By default, `this` is the global object.
+
+### Symbolicating
+```lisp
+(coerce-string)  ;; ( value -- string* )
+(coerce-symbol)  ;; ( value -- symbol )
+```
+
+### Object Manipulation
+```lisp
+(new-object)        ;; ( -- object )                    push a new empty object reference to stack
+(push-from-object)  ;; ( slot-symbol object -- value )  indirect object load
+(pop-into-object)   ;; ( value slot-symbol object -- )  indirect object store
+```
+
+### Call
+```lisp
+(call-function fn-index nargs)  ;; ( arg* -- result )
+(call-primitive prim-index)     ;; ( arg* -- 0 )
+(push-function fn-index)        ;; ( -- fn-object )            push fn-object
+(call-object nargs)             ;; ( args* object -- result )  pop object and call as function
+```
+
+### Arithmetic
+```lisp
+(add)
+(subtract)
+(multiply)
+(divide)       ;; ( lhs rhs -- result )
+(power)        ;; ( x y -- result ) 
+(mod)          ;; ( x y -- r )
+(shift-left)   ;; ( nbits value -- result )
+(shift-right)  ;; ( nbits value -- result )
+(zshift-right) ;; ( nbits value -- result )
+(bit-not)
+(bit-and)
+(bit-or)
+(bit-xor)
+```
+
+* Throws TypeError on badly typed input
+
+### Boolean
+```lisp
+(coerce-bool)  ;; ( value -- bool )
+(not)     ;; ( value -- bool )
+(and)     ;; ( value value -- bool )
+(or)      ;; ( value value -- bool )
+(xor)     ;; ( value value -- bool )
+```
+
+Boolean coercion can be explicitly performed, for equality tests against true.
+Boolean operators implicity coerce their values to booleans.
+
+The mapping is:
+
+* True: object, non-empty-string, non-zero integer or double
+* False: null, false, undefined, 0, ""
+
+
+### Untyped Comparison
+```lisp
+(equal)        ;; ( lhs rhs -- bool )
+(not-equal)
+(less-than)
+(greater-than)
+(greater-or-equal-than)
+(less-or-equal-than)
+```
+
+* Note that equal does not coerce values.
+
+### Jump Operators
+```lisp
+(jmp target)
+(jmp-if target)      ;; ( bool -- )
+```
+
+* Pop the top value of the stack and compare to true.
+* Implicitly coerces the value to a boolean, as if (coerce-bool).
+
+### Primitive Reflection
+```lisp
+(instance-of kind)  ;; ( thing -- bool )
+```
+
+where kind is one of:
+
+| Code | Mmemonic |
+|------|----------|
+| 0    | object   |
+| 1    | integer  |
+| 2    | double   |
+| 3    | symbol   |
+
+### Misc Operators
+```lisp
+(nop)               ;; ( -- ) the best instruction
+(breakpoint id)     ;; ( -- ) drop out to interpreter debug handler
+(exit exit-code)    ;; ( -- ) terminate.
+(throw)             ;; ( value -- ) Pop value and throw out to handler.
+```
+## Extension: Exception handling
+see: https://github.com/WebAssembly/exception-handling/blob/master/proposals/Level-1.md
+
+## Extension: Atomics?
+see: https://llvm.org/docs/Atomics.html
+
+## Extension: Inline caches
+see: https://jandemooij.nl/blog/2017/01/25/cacheir/
+
+# Values in b9
```

Comment:
Maybe want to move this section to the top of the document?

Reply:
Sure, will do.